src/vr/clock.zig

```diff
@@ -74,328 +75,337 @@ const Epoch = struct {
     }
 };
 
-pub const Clock = struct {
-    const Self = @This();
-
-    allocator: *std.mem.Allocator,
-
-    /// The index of the replica using this clock to provide synchronized time.
-    replica: u8,
-
-    /// The underlying time source for this clock (system time or deterministic time).
-    time: *Time,
-
-    /// An epoch from which the clock can read synchronized clock timestamps within safe bounds.
-    /// At least `config.clock_synchronization_window_min_ms` is needed for this to be ready to use.
-    epoch: Epoch,
-
-    /// The next epoch (collecting samples and being synchronized) to replace the current epoch.
-    window: Epoch,
-
-    /// A static allocation to convert window samples into tuple bounds for Marzullo's algorithm.
-    marzullo_tuples: []Marzullo.Tuple,
-
-    /// A kill switch to revert to unsynchronized realtime.
-    synchronization_disabled: bool,
+pub fn ClockImpl(comptime ClockType: type) type {
+    return struct {
+        const Self = @This();
```

Comment:
Hmm, I've already made essentially this change to make `Clock` generic over `Time` on the `beta2` branch. Not sure what the best way to resolve this conflict would be.